Code/Framework/AzCore/AzCore/IO/Path/Path.h

```diff
@@ -398,12 +398,13 @@ namespace AZ::IO
         // appends
         //! Joins a path together using the following rule set
         //! let p = other PathView
-        //! # If p.is_absolute() || (p.has_root_name() && p.root.name() != root_name()
+        //! # If p.is_absolute() || p.starts_with_alias() || (p.has_root_name() && p.root.name() != root_name()
         //! Then replaces the current *this path with p
         //! NOTE: This is the case of trying to append something such as
         //! "C:\Foo" / "F:\Foo" = "F:\Foo"
         //! IMPORTANT NOTE: this also applies for the root directory as well
-        //! So appending "test/foo" "/bar" results in "/bar", not "test/foo/bar"
+        //! So appending "test/foo" "/bar" results in "/bar", not "test/foo/bar",
+        //! also appending "test/foo" "@products@/bar" results in "@products@/bar", not "test/foo/@products@/bar".
```

Comment:
This fix was necessary, VFS doesn't work without it because there are places in the code where it's doing a path append with an absolute path and it needs to not be appended in the same way when using aliases.

Example:
In Launcher.cpp there is this lines
````
auto autoExecFile = AZ::IO::FixedMaxPath{pathToAssets} / "autoexec.cfg";
AZ::Interface<AZ::IConsole>::Get()->ExecuteConfigFile(autoExecFile.Native());
````
`autoExecFile` has an absolute path `/absolute/path/to/autoexec.cfg` when calling `ExecuteConfigFile`. This eventually reaches `MergeSettingsToRegistry_ConfigFile` and the first thing it does is:
````
auto configPath = FindProjectRoot(registry) / filePath;
````
So `FindProjectRoot(registry)` returns an absolute path, `filePath` is an absolute path too, the result `configPath` is not the concatenation of both, but `filePath`.

So the same has to happen when aliases are used.
Because if `FindProjectRoot(registry)` returns `@projectroot@` and filePath is `@products@/autoexec.cfg`, the result needs to be `"@products@/autoexec.cfg"` and not the concatenation `"@projectroot@/@products@/autoexec.cfg"` (which will result is a bad path once aliases are resolved).

Reply:
This change doesn't belong in that Path class.
The File Alias system is at higher level level than the low level AZ::IO::Path class.

The FileIO system is what is responsible for resolving paths before any filesystem path manipulation takes place

Reply:
After talking to @lumberyard-employee-dm I'm going to try a different approach, using `/@alias@` in the paths instead of `@alias@`, that way Path class will pick it up correctly. 

I tried that before and it didn't work because remote system couldn't resolve it correctly locally, but there are a couple of things I can try to get that working:
- Make modifications to Local IO Resolve Path function to support `/@alias@`
- Or if Remote IO has its own resolve, it could just remove the `/` part and send it to the remote system without the `/`.

I'll try the second approach first as it's easy, see if it all works and then I might try to have a go to the first approach and see how it goes (it's a more impactful change for all the other platforms to change Local IO Resolve Path function).

Reply:
Done. The VFS now sets the paths with `/@alias@` and `LocalFileIO::ResolvePath` can handle it. This changes to the Path class have been reverted.